pkgs/build-support/rust/default.nix

```diff
@@ -102,6 +102,7 @@ stdenv.mkDerivation ((removeAttrs args ["depsExtraArgs"]) // lib.optionalAttrs u
   inherit buildAndTestSubdir cargoDeps;
 
   cargoBuildType = buildType;
+  cargoBuildProfileFlag = lib.optionalString (buildType != "debug") "--${buildType}";
```

Comment:
This should IMO not be done in `default.nix`, because then everyone who uses one of the hooks needs to define  `cargoBuildProfileFlag`. The hooks that use `cargoBuildType` (`cargoBuildHook` and `cargoCheckHook`) should contain the logic to add a flag when `cargoBuildType != debug`.

Reply:
Oh!!! Okay! I get it! Thanks for this information! I think I've done this now!

Reply:
Thanks a lot for making the changes. It seems that there is still a merge conflict against staging.

(This change needs to go through staging, since it results in a lot of package rebuilds.)